### PR TITLE
Base model/domain classes for Bindings API.

### PR DESCRIPTION
**Pull request by tmacam**
PENDING https://github.com/dapr-sandbox/components-java-sdk/pull/25 

# Description

* Updates code to also download bindings.proto and generate
  Java code for it.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: components-java-sdk/src/main/java/io/dapr/components/domain/bindings/OutputBinding.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   * @return Mono holding a {@link InvokeResponse} with the result of this invoke call.
+   */
+  Mono<InvokeResponse> invoke(InvokeRequest request);
+
+  /**
+   * ListOperations list system supported operations.
+   *
+   * @return A Mono with the List of operations suported by this output binding component.
+   */
+  Mono<ListOperationsResponse> listOperations();
```
Comment (philliphoff) [inline components-java-sdk/src/main/java/io/dapr/components/domain/bindings/OutputBinding.java:39]: Consider "unwrapping" the response by requiring only the inner operations list be returned?

**Review by philliphoff — APPROVED** [review]
Just one question, but otherwise LGTM.

File: components-java-sdk/src/main/java/io/dapr/components/domain/bindings/OutputBinding.java (hunk 1 above)
Comment (tmacam) [reply components-java-sdk/src/main/java/io/dapr/components/domain/bindings/OutputBinding.java]: Good point. Done on 44c9a67fe49f65664a9a93e0b5b13b55a6d8ae03

**Comment by tmacam** [comment]
Rebased just to remove what was already merged in #25

**Pull request merged**